src/enclave-agent/src/config.rs

```diff
@@ -0,0 +1,126 @@
+use serde::{Deserialize, Serialize};
+use std::{fs::File, io::Write, path::Path};
+
+const DEFAULT_KEYPROVIDER_CONFIG_PATH: &str = "/etc/ocicrypt_keyprovider_native.conf";
+const DEFAULT_KEYPROVIDER_CONFIG: &str = r#"
+{
+    "key-providers": {
+        "attestation-agent": {
+            "native": "attestation-agent"
+        }
+    }
+}
+"#;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct DecryptConfig {
+    pub key_provider: Option<String>,
+    #[serde(default = "default_security_validate")]
+    pub security_validate: Option<bool>,
+    #[serde(default = "default_auth")]
+    pub auth: Option<bool>,
+}
+
+fn default_security_validate() -> Option<bool> {
+    Some(true)
+}
+
+fn default_auth() -> Option<bool> {
+    Some(false)
+}
+
+impl DecryptConfig {
+    pub fn load_from_file(file_path: &String) -> Result<Self, std::io::Error> {
+        let file = File::open(file_path)?;
+        let config: DecryptConfig = serde_json::from_reader(file)?;
+        Ok(config)
+    }
+}
+
+pub struct KeyproviderConfig {
+    pub config_path: String,
+    pub is_exist: bool,
+}
+
+impl KeyproviderConfig {
+    pub fn default_config() -> Result<Self, std::io::Error> {
+        let path = DEFAULT_KEYPROVIDER_CONFIG_PATH.to_string();
+        let mut kc = Self {
+            config_path: path.clone(),
+            is_exist: Path::new(&path).exists(),
+        };
+
+        if !kc.is_exist {
+            File::create(Path::new(&path))?.write_all(DEFAULT_KEYPROVIDER_CONFIG.as_bytes())?;
+            kc.is_exist = true;
+        }
+        Ok(kc)
+    }
+
+    pub fn export_to_env(&self) {
+        std::env::set_var("OCICRYPT_KEYPROVIDER_CONFIG", self.config_path.clone());
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn test_load_decrypt_config() {
```

Comment:
I think this is not testing what we are interested in:  errors are correctly returned when file read or json parsing fails based on inputs provided.

Reply:
The rust standard create provides `file read` APIs, it is out of the responsibilities to cover all file read failures.

The crate `serde-json` provides `json parsing` APIs. It will return different error info once reading invalid json files, it is hard to emulate all cases.

I think we at most add two cases, including reading a not existing file, reading a json files with invalid key, to represent file read failure and json parse failure.

Reply:
> I think we at most add two cases, including reading a not existing file, reading a json files with invalid key, to represent file read failure and json parse failure.

makes sense